Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -74,6 +72,11 @@ public abstract class ClientBuilderBase<TClient>
         /// </summary>
         public string UserAgent { get; set; }
 
+        /// <summary>
+        /// The gRPC implementation to use, or null to use the default channel factory.
+        /// </summary>
+        internal GrpcImplementation GrpcImplementation { get; set; }
```

Comment:
I believe `protected abstract GetChannelPool()` might need to use this if it ends up creating the pool, so this one should probably be `protected internal`?
I see that GrpcImplementation is still internal by design, but still the possible discrepancy between this one here and the one in ChannelPool, I don't like.

I wonder if it would be possible to have something like GrpcImplementation.Current, that falls back to GrpcImplementation.Default, but that can be set on startup (and maybe even just once) so that everything can refer back to that and we don't need to be passing a GrpcImplementation around?

Reply:
The intention is that the public API doesn't need to change right now. I'm not sure what you mean by "the possible discrepancy between this one here and the one in ChannelPool" - the idea is that they both go public at the same time, when we're slightly more confident about the shape. (We *could* make it public right now, but I don't think we need to just yet.)

Reply:
Yes, I'm (mostly*) fine with them not being public right now. What I meant by "the possible discrepancy..." is that, as they stand, for any given execution, this one could bet set to one implementation and the one in `ChannelPool` to another implementation which I think is not needed/wanted and might be problematic. That's why I was thinking of having a `static` `GrpcImplementation.Current` (as in current execution implementation) that both client builders and `ChannelPool` could use, without the need of passing a `GrpcImplementation` around. `GrpcImplementation.Current` can be set on startup (can be set only once probably) and if not set would fallback to `GrpcImplementation.Default`.

\* Mostly fine because of the `protected abstract GetChannelPool()` method in `ClientBuilderBase` should be able to access `ClientBuilderBase.GrpcImplementation` in case it ends up building a channel. But given that everything is internal this is just a nit.